### PR TITLE
add endpoint for listing connections and creating internal accounts

### PR DESCRIPTION
**Pull request by yazan-equityzen**

### CODE REVIEW

File: modern_treasury/modern_treasury.py [hunk 1]
```diff
@@ -238,4 +241,19 @@ def list_incoming_payment_detail(self, virtual_account_id: str = None) -> List[O
             response = self._get(url=LIST_INCOMING_PAYMENT_DETAIL_URL, params=querystring)
             return [IncomingPaymentDetailResponse(payment_order) for payment_order in response]
         except:
-            return []
\ No newline at end of file
+            return []
+    
+    def list_connections(self, vendor_customer_id: Optional[str] = None, entity: Optional[str] = None) -> List[ConnectionResponse]:
```
Comment (ez-issam) [inline modern_treasury/modern_treasury.py:246]: do we need the idempotency key for a GET?

File: modern_treasury/modern_treasury.py [hunk 2]
```diff
@@ -9,9 +9,11 @@
 from modern_treasury.objects.request.expected_payment import ExpectedPaymentRequest
```
Comment (ez-issam) [inline modern_treasury/modern_treasury.py:9]: I would add another method to be able to choose the connection we want.
Here is an example response from the connections list:

```
[{"id":"289cd2f0-20bf-4f73-98ae-1535f90f31e0","object":"connection","live_mode":false,"vendor_id":"example1","vendor_name":"Gringotts Wizarding Bank","vendor_customer_id":null,"discarded_at":null,"created_at":"2021-05-19T14:39:28Z","updated_at":"2021-05-19T14:39:28Z"},{"id":"55a000f7-6eb1-42ff-a160-6ed3c1087b60","object":"connection","live_mode":false,"vendor_id":"example2","vendor_name":"Iron Bank of Braavos","vendor_customer_id":null,"discarded_at":null,"created_at":"2021-05-19T14:39:28Z","updated_at":"2021-05-19T14:39:28Z"}]```

We should be able to grab the connection based on the vendor_name and vendor_id

File: modern_treasury/modern_treasury.py (hunk 1 above)
Comment (ez-issam) [reply modern_treasury/modern_treasury.py:246]: Also please add some tests 🙇

Comment (yazan-equityzen) [reply modern_treasury/modern_treasury.py:246]: > do we need the idempotency key for a GET?

Nope, we don't need idempotency key for GET requests

**Review by ez-issam — APPROVED** [review]

**Pull request merged**